pageserver/src/pgdatadir_mapping.rs

```diff
@@ -854,6 +854,49 @@ impl Timeline {
     }
 }
 
+pub const AUX_FILES_V2: bool = true;
+
+fn split_prefix<'a, 'b>(path: &'a str, prefix: &'b str) -> Option<&'a str> {
+    if path.starts_with(prefix) {
+        Some(&path[prefix.len()..])
+    } else {
+        None
+    }
+}
+
+fn hash_to_raw_key(prefix: u16, data: &[u8]) -> [u8; RAW_KEY_SIZE] {
+    let mut key = [0; RAW_KEY_SIZE];
+    let hash = crc32c::crc32c(data).to_be_bytes();
+    let prefix = prefix.to_be_bytes();
+    key[0] = prefix[0];
+    key[1] = prefix[1];
+    key[2] = hash[0];
+    key[3] = hash[1];
+    key[4] = hash[2];
+    key[5] = hash[3];
```

Comment:
need some hashing algorithm to generate more bytes (farmhash?)

Reply:
What are the pros for farmhash? Doing a quick scan around smhasher repositories ([rurban](https://github.com/rurban/smhasher?tab=readme-ov-file#summary), [fwojcik](https://gitlab.com/fwojcik/smhasher3/-/blob/main/results/README.md#passing-hashes)) ahash and xxh3 are still valid. Perhaps ahash would even be preferable given it's usage in rust ecosystem and the acceleration we will most likely always have. (Also, already a dependency of ours.)

However ahash docs remind:

> Because aHash does not have a fixed standard for its output, it is able to improve over time. But this also means that different computers or computers using different versions of ahash may observe different hash values for the same input.

As we really don't want to be platform dependent (amd64 vs. arm64) I think this becomes a question of do we want stable hashes or do we want the fastest hash?

If we want stable hashes, truncating cryptographic standard hashes would probably be the best option.

If we can have the hash evolve, then using ahash would probably be the better option.

Reply:
Any hash algorithm would be fine and farmhash is the first one popped out of my head 🤣

Reply:
Okay, I think the stability aspect is important for us. I am a bit skeptical about whether it's feasible to just define a test for many keys and hope to have the fixed hashes for them because any upstream would not accept the bug unless the hash function was cryptographic.

Stability might be an already established term for hash functions returning the same value on different platforms/targets and across versions. Of course, we could still get hit by a bug where we upgrade to a regressed version, but that just needs to be hardcoded.